script.js

```diff
@@ -1 +1,362 @@
+      // create the square element
+      const square = document.createElement('div');
+
+      // set a class for CSS purposes
+      square.classList.add('square');
+
+      // set the click event
+      // eslint-disable-next-line
+      square.addEventListener('click', (event) => {
+        // we will want to pass in the card element so
+        // that we can change how it looks on screen, i.e.,
+        // "turn the card over"
+        squareClick(event.currentTarget, i, j);
+      });
+
+      rowElement.appendChild(square);
+    }
+    boardElement.appendChild(rowElement);
+  }
+
+  return boardElement;
+};
+
+const getName = () => {
+  messageDisplay('Please input your name :)');
+
+  const userNameInput = document.createElement('input');
+  userNameInput.setAttribute('placeholder', 'Your Name');
+
+  const submitButton = document.createElement('button');
+  submitButton.innerText = 'Submit';
+  submitButton.addEventListener('click', () => {
+    userName = userNameInput.value;
+    // Remove the name input and submit buttons.
+    userNameInput.remove();
+    submitButton.remove();
+    // eslint-disable-next-line no-use-before-define
+    initGame();
+  });
+
+  document.body.appendChild(gameInfo);
+  document.body.appendChild(userNameInput);
+  document.body.appendChild(submitButton);
+};
+
+const gamePlay = () => {
+  // Set up a timer for 3mins. Once 3mins is up, the board will disappear.
+  // Countdown Timer displays remaining time in game.
+  let gameTime = 180000;
+  let isPaused = false;
+  const delayInMillisceonds = 1000;
+  const timerDisplay = document.createElement('div');
+  timerDisplay.classList.add('timer-display');
+
+  const ref = setInterval(() => {
+    const secondDisplay = (gameTime / 1000) % 60;
+    const minuteDisplay = ((gameTime / 1000) - secondDisplay) / 60;
+    timerDisplay.innerText = `Time Remaining: ${minuteDisplay}:${secondDisplay}`;
+
+    if (isPaused !== true) {
+      if (gameTime <= 0) {
+        clearInterval(ref);
+      }
+      gameTime -= 1000;
+
+      if (gameTime === 0 && totalWin === 0) {
+        // eslint-disable-next-line no-use-before-define
+        boardEl.innerHTML = '';
+        messageDisplay(`Sorry ${userName}... Time's up! Better luck next time!`);
+      } else if (gameTime === 0) {
+        // eslint-disable-next-line no-use-before-define
+        boardEl.innerHTML = '';
+        messageDisplay(`Thanks for playing ${userName}! Time's up! You won ${totalWin} times! Nice!`);
+      }
+    }
+  }, delayInMillisceonds);
+
+  const pauseButton = document.createElement('button');
+  pauseButton.classList.add('timer-button');
+  pauseButton.innerText = 'Pause';
+  pauseButton.addEventListener('click', () => {
+    isPaused = true;
+  });
+
+  const resumeButton = document.createElement('button');
+  resumeButton.classList.add('timer-button');
+  resumeButton.innerText = 'Start';
+  resumeButton.addEventListener('click', () => {
+    isPaused = false;
```

Comment:
consider adding more comments so that it is easier for the reader to follow your code